types/react-router/v3/react-router-tests.tsx

```diff
@@ -123,19 +123,39 @@ class UserList extends React.Component<UserListProps & WithRouterProps> {
 
 const UserListWithRouter = withRouter(UserList);
 
+interface AvatarProps {
+    user: string;
+}
+
+const Avatar: React.FunctionComponent<AvatarProps & WithRouterProps> = ({ user, children, location, params, router, routes }) => (
```

Comment:
[PP] `FC` can be used instead of `FunctionComponent`

```suggestion
const Avatar: React.FC<AvatarProps & WithRouterProps> = ({ user, children, location, params, router, routes }) => (
```

Reply:
Please don't use abbreviations in public types. It's ok for you to use them internally but these types should be as accessible as possible. The abbreviation wouldn't serve any purpose.

Reply:
Per response I'll leave it as `FunctionComponent`. As an aside, there are references to the deprecated `StatelessComponent` throughout this definition, I was unsure whether to use `StatelessComponent` here for consistency or replace all `StatelessComponent` with `FunctionComponent`?

Reply:
Makes sense to stay with `FunctionComponent`, agree with that.

Maybe a follow up PR could fix the usage of `StatelessComponent`. Wdyt?